crates/provider-sdk/src/provider_main.rs

```diff
@@ -119,17 +119,39 @@ where
     })?;
     let connection = get_connection();
 
-    // pre-populate provider and bridge with initial set of link definitions
-    // initialization of any link is fatal for provider startup
+    // Pre-populate provider and bridge with initial set of link definitions
+    // Initialization of any link is fatal for provider startup
     let initial_links = host_data.link_definitions.clone();
     for ld in initial_links.into_iter() {
-        if !provider.put_link(&ld).await {
-            error!(
-                link_definition = ?ld,
-                "Failed to initialize link during provider startup",
-            );
-        } else {
-            connection.put_link(ld).await;
+        // Get config for the link name
+        //
+        // todo(vados-cosmonic): this doesn't make sense -- are named configs included in the link def
+        // supposed to be looked up by the provider when they're received? Why do we get an all combined one
+        // that is keyed by link name? (and if it wasn't keyed by link name it'd be all smushed together)
```

Comment:
We shouldn't be giving an all combined one, what we should be giving the provider at initialization time is a list of link configurations that includes the link name, the source ID and the target and based on the source we should deliver them to the provider for handling. It's up to the provider to store the config as it needs to

Reply:
Ah see the note earlier, got it! I've reverted the `configs` change